_posts/2017-01-17-Oredev2016.md

```diff
@@ -0,0 +1,695 @@
+Think again, he can breakout and he can gain control of the whole operating system.
+
+
+## Building secure images
+
+* Choose base image carefully
+* Strategy for security patches - Regularly update your Docker container so it gets the necessary updates
+* Continuously rebuild and test - Security patches in your build environment
+* In a secure build environment
+* Digitally sign docker images - `# docker build -t test --disabled-content-trust=false`
+
+
+## Internal private registry - Push and pull the containers internally
+This private registry is insecure by default meaning.
+If an attacker gains control of the network where the registry is located, he can easily read and modify each image.
+An attacker pushes a modified image and the orchestration will pull this.
+
+If the attacker gains control of your CI/CD server he can again push modified images to the registry.
+To avoid this attack, be sure to have these activated inside your registry.
+
+* Secure transfer using TLS
+* Authenticate Users
+* Limit access rights
+* Docker Content Trust - will digitally sign your images
+* Secure your development environment
+
+## How does the attacker inject code?
+>You use whatever docker image that is in the registry and add your backdoor binary in a Dockerfile,
+then you build the image and point it to the very same image in the registry
+
+## Security measurements inside the container
+* User namespaces
+An extra security measure are user namespaces, this will make the root in the container differentiate from the root in the host.
+* Remote API
+If you have no good reason for using the remote API then you should disable this option.
+If you have to use it, be sure to use secure transport over TLS and that you have client authentication so you don't let the world in.
+* Try enable SELinux (enforcing) on the host
+* Drop unused capabilities that are installed by default and install the capabilities that you actually need.
+    * `--cap-drop=ALL`
+    * `--cap-add=needed_cap `
+* Only pull from private registries
+
+## Things you should avoid in Docker
+* --privileged  = breakout is trivial and will give you access to different machines
+    * Be sure to check that Kubernetes is not doing this
+* --insecure-registry = disable tls requirement
+* --net=host = use host network stack
+
+
+<a name="kubernetes" />
+
+# Kubernetes automation in production - Paul Bakker
+
+<img class="float-image p-image" src="{{ '/img/oredev/paulbakker.jpg' | prepend: site.baseurl }}" alt="Paul Bakker" style="width: 100px"/>
+
+[Paul Bakker](http://www.twitter.com/pbakker) is working as a software architect for Luminis Technologies, where he's currently leading the team that develops the CloudRTI
+a Kubernetes/Docker based service that takes care of DevOps tasks like blue/green deployment, failover, centralised logging and monitoring.
+He is the author of “Building Modular Cloud Apps With OSGi” published by O'Reilly and an advocate for modularity in software.
```

Comment:
MS Word double quotes, should also be replaced by regular quotes.
Would leave these out though and instead just have it link to the book.